### DISCUSSION TITLE
Difference between dev and codespace

### DISCUSSION THREAD

**Discussion by Praskand** [community/community#138001] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

![image](https://github.com/user-attachments/assets/bab70b5f-703f-49f5-aeff-24b563bf073f)
On hover it says: JSX element class does not support attributes because it does not have a 'props' property.ts(2607)
This error does not occur in codespace and is completely fine. Why?

**Comment by SjxSubham** [comment]
This typically occurs when TypeScript is unable to recognize the props property on a JSX element. This discrepancy between your Git development environment and Codespaces could be due to differences in TypeScript configurations or versions.

1. Check TypeScript version
2. Update Typescript
3. check `tsconfig.json` Compare the `tsconfig.json` files in both environments. Ensure that the configurations are the same, especially the `jsx` and `strict` settings. Here is an example of a typical `tsconfig.json` :  

```
{
  "compilerOptions": {
    "target": "es5",
    "module": "commonjs",
    "jsx": "react",
    "strict": true,
    "esModuleInterop": true,
    "skipLibCheck": true,
    "forceConsistentCasingInFileNames": true
  }
}
```
4. Ensure that all dependencies, including @types/react and @types/react-dom, are the same in both environments. You can check the versions in package.json and update them if necessary
```
npm install @types/react@latest @types/react-dom@latest
```

_By ensuring consistency in TypeScript versions, configurations, and dependencies, you should be able to resolve the discrepancy between your Git development environment and Codespaces. Let me know if you need further assistance!_

## Mark my comment as answered, If ur doubt resolved...

**Reply by Praskand** [reply]
ts 5.5.4
both have same json files. I think its problem with dev

**Comment by UvejsMikullovci** [comment]
just burn the FUCKING COMPUTER

**Comment by manukumar7** [comment]
You're seeing the error locally because your tsconfig.json likely has "jsx": "react" instead of "react-jsx". This makes TypeScript expect class-based components, but react-movable exports function components.

✅ Fix: In tsconfig.json, change:
"jsx": "react-jsx"
Then restart your editor. That’s why it works fine in Codespaces—they use the correct JSX setting.

**Comment by YadurajManu** [comment]
The error occurs because your class component doesn’t define a `props` type, so TypeScript doesn't know what attributes it can accept. In Codespaces, it works fine likely because it uses a different TypeScript version or a more relaxed `tsconfig.json`, which may skip strict checks. To fix it, explicitly define the props like `class MyComponent extends React.Component<{ src: string }>` or use a function component, which is simpler and often preferred in modern React.

**Comment by Aashik9567** [comment]
✅ Why it doesn't happen in Codespaces:
Codespaces often has:

Newer TypeScript version with better JSX inference.

More up-to-date tsconfig.json or type definitions.

Slightly different module resolution or project settings than your local setup.

So, the error might not appear there due to more forgiving or modern configuration.

🔧 How to Fix It Locally
You likely have a component like this:

tsx
Copy
Edit
class ImageComponent extends React.Component {
  render() {
    return <img src="..." />;
  }
}
But if you try to use:

tsx
Copy
Edit
<ImageComponent src="..." />
TypeScript expects ImageComponent to define props.

✅ Fix:
Add an explicit props type — even if it's just {}:

tsx
Copy
Edit
type Props = {
  src: string;
  alt?: string;
};

class ImageComponent extends React.Component<Props> {
  render() {
    return <img src={this.props.src} alt={this.props.alt} />;
  }
}
Or switch to a functional component (which has better inference by default):

tsx
Copy
Edit
const ImageComponent = ({ src, alt }: { src: string; alt?: string }) => (
  <img src={src} alt={alt} />
);
🧠 Tip:
Make sure your local:

TypeScript version is up-to-date (npm install typescript@latest)

tsconfig.json includes correct jsx, module, and target settings.

Let me know if you want help debugging your tsconfig or upgrading TypeScript!
